### ISSUE TITLE
prompt doesn't always update

### ISSUE THREAD

**Issue by ohAitch** [urbit/urbit#353]
upon the pier first starting, after ~&s and idle ^C, etc.

**Comment by galenwp** [comment]
Is there a way to actually cause this from `dojo>`?  I remember this being an issue, but haven't seen it recently.

**Comment by cgyarvin** [comment]
This is an ancient bug which I don't think exists anymore.

**Comment by ohAitch** [comment]
The way to cause this from `dojo>` is to hit `^C` without anything to interrupt, printing

```
~zod:dojo> 
ignored: intr

interrupt

```

with no new prompt. Similarly if you cause an e.g. `%mo-not-running` printf by running `:~zod/nonexistent ~` on a different ship, the prompt will not refresh afterwards.

**Comment by juped** [comment]
Yeah, we decided to leave this one open but didn't comment, thanks for reopening it.

**Comment by matildepark** [comment]
@joemfb @ohAitch Can't reproduce this, why is this still open?

**Comment by joemfb** [comment]
`|mass` on `ipc-redux` is a recent example.

It'd be hard to say this is completely fixed without substantial work on the terminal subsystem. FWIW, a good portion of that has already been done in king haskell ...

**Comment by Fang-** [comment]
Can no longer reproduce with `|mass`, or any of the previous variations. If anyone still has repro steps for this, happy to pick up as part of my terminal work. Otherwise, this should get closed soon.

**Comment by joemfb** [comment]
Writing to `stderr` (as is often done in the mars/serf process) can still clobber the prompt. Fully fixing this requires a real logging system.

**Comment by zalberico** [comment]
> Fully fixing this requires a real logging system.

Because then logging will reduce the load on the prompt? If so I'd like to make an issue for the logging system (if it doesn't already exist) that Alex is going to start on soon and we can then close this (linking it in the description of the logging system issue first). Basically turning this into something more actionable we can complete to fix this.
